ts/components/dialog/SessionProInfoModal.tsx

```diff
@@ -259,3 +261,18 @@ export const showSessionProInfoDialog = (
     })
   );
 };
+
+export const useShowSessionProInfoDialogCb = (variant: SessionProInfoVariant) => {
+  const dispatch = useDispatch();
+  const isProAvailable = useFeatureFlag('proAvailable');
+  const mockHasPro = useFeatureFlag('mockUserHasPro');
+
+  // TODO: get pro status from store once available
+  const hasPro = mockHasPro;
+
+  if (!isProAvailable || hasPro) {
+    return () => null;
+  }
+
+  return () => showSessionProInfoDialog(variant, dispatch);
+};
```

Comment:
that's fine here, but I have been creating those in `ts/components/menuAndSettingsHooks`.
Not really relevant for this one though ?

Reply:
hm yeah think it works better here as its specific to opening this dialog